### ISSUE TITLE
Stuck after 1014 requests.

### ISSUE THREAD

**Issue by whizsid** [Project-OSRM/osrm-backend#5542]
I have built the OSRM backend on my ubuntu 18 server by following [these](https://github.com/Project-OSRM/osrm-backend/wiki/Building-OSRM#building) steps. After that, I have made 60,000 different requests by a program. After 1014  requests processed, OSRM server stuck and did not give me a response via curl. Neither printed my request with OSRM. 

But when I build the same source with `DCMAKE_BUILD_TYPE=Debug`  All requests succeeded.

**Comment by akashihi** [comment]
Did you used keepalive connections?

**Comment by whizsid** [comment]
No. Just a fresh installation. This is my curl options set by a PHP script.

```
       CURLOPT_RETURNTRANSFER => 1,
        CURLOPT_URL => "http://localhost:5000/route/v1/driving/$lng1,$lat1;$lng2,$lat2?overview=false&alternatives=true&steps=true",
        CURLOPT_USERAGENT => 'Salesman Alogythm Request'

```

**Comment by akashihi** [comment]
I just made ~10k route requests on a same osrm built from latest master without any issues.

**Comment by Nate-Wessel** [comment]
I experienced a similar problem. I didn't count the exact number of requests made, but it was right around 1k. I reproduced this on two different computers, also running ubuntu 18.04. 
At first, I was running some older code on one computer (maybe 2-3 months out of date) and tried to fix this by updating and rebuilding the latest OSRM version. Same issue in both the older and latest versions.

Killing the server and restarting it would give me another ~1k requests before the same timeout errors appeared again - the server was not responding.

When I found this issue, I tried rebuilding with debug mode as suggested and found that everything worked as expected.
